lib/linguist/languages.yml

```diff
@@ -6480,6 +6480,15 @@ cURL Config:
   tm_scope: source.curlrc
   ace_mode: text
   language_id: 992375436
+browserslist:
+  type: data
+  aliases:
+  - browserslist
+  filenames:
+  - ".browserslistrc"
```

Comment:
**question:** do i need to specify a file `browserlist` without `rc` and dot before filename here ? Or is it enough to specify it in `aliases` ?

Reply:
@voskresla Treat all permutations as separate filenames (provided they meet the usage requirement). That is, like this:

~~~yaml
filenames:
- ".browserslist"
- ".browserslistrc"
- "browserslist"
- "browserslistrc"
~~~

As for `aliases`, they're only used to specify the name of a language in `.gitattributes` files and fenced code-blocks—they have nothing to do with how/what Linguist matches by way of filenames or extensions. For example, [JavaScript has `node`](https://github.com/github/linguist/blob/7c2adbdb15d4efd25d92cd5ed20d0025f0d32d28/lib/linguist/languages.yml#L2593) as an alias, so something like this (normally) receives JS highlighting:

~~~markdown
```node
import {getWindowSize, env, platform} from "./index.mjs";

console.log(getWindowSize());

console.log("gsdgds" in env);
env.gsdgds = 1;
console.log("gsdgds" in env);
```
~~~

The above block isn't showing the highlighted Markdown source, because our Markdown grammar happens to be kinda crap (then again, it's rarely seen or used, so… 🤷‍♂️)

Reply:
㈽ thx for explanation.

Reply:
> ㈽

[What the](https://graphemica.com/%E3%88%BD)… 😀